addons/mail/models/mail_activity.py

```diff
@@ -164,6 +172,50 @@ def action_done(self, feedback=False):
     def action_close_dialog(self):
         return {'type': 'ir.actions.act_window_close'}
 
+    @api.multi
+    def unlink(self):
+        for activity in self:
+            notification = {'type': 'activity_updated', 'activity_deleted': True}
+            self.env['bus.bus'].sendone((self._cr.dbname, 'res.partner', activity.user_id.partner_id.id), notification)
+        return super(MailActivity, self).unlink()
+
+    @api.model
+    def get_activities(self):
+        value = []
+        query = """SELECT m.name,count(*), a.res_model as model, a.res_id as res_id,
+                        CASE
+                            WHEN now()::date - a.date_deadline::date = 0 Then 'today'
+                            WHEN now()::date - a.date_deadline::date > 0 Then 'overdue'
+                            WHEN now()::date - a.date_deadline::date < 0 Then 'planned'
+                        END AS states
+                    FROM mail_activity AS a
+                    JOIN ir_model AS m ON a.res_model = m.model
+                    WHERE user_id = %s
+                    GROUP BY m.name,states, a.res_model,a.res_id;
```

Comment:
@tivisse , In a query it's not possible to group by on one field, whatever we are going to select must appear in the GROUP BY clause or be used in an aggregate function in postgres. otherwise,  it will give you an error. Thanks